### PR TITLE
Feature/improve attachment

### LINKED ISSUE #247: Create delete attachment method

### LINKED ISSUE #248: Create service for working with AzureStorageBlobs

### LINKED ISSUE #249: Add validation into attachment service

### PR DESCRIPTION
**Pull request by SnipeFrost**
1. Moved logic for working with AzureStorageBlobs in other service
2. Created delete attachment method
3. Added validation into attachment service

### CODE REVIEW

File: CharlieBackend.Business/Services/AttachmentService.cs [hunk 1]
```diff
@@ -116,5 +106,69 @@ public async Task<Result<DownloadAttachmentDto>> DownloadAttachmentAsync(long at
...
+            await _unitOfWork.AttachmentRepository.DeleteAsync(attachmentId);
+
+            await _unitOfWork.CommitAsync();
+
+            return Result<AttachmentDto>.GetSuccess(_mapper.Map<AttachmentDto>(attachment));
+        }
+
+        public bool AttachmentsExtentionValidation(IFormFileCollection fileCollection)
+        {
+            string[] dangerousExtentions =
```
Comment (SerjKocherga) [inline CharlieBackend.Business/Services/AttachmentService.cs:131]: Make it const or readonly.

File: CharlieBackend.Business/Services/AttachmentService.cs [hunk 2]
```diff
@@ -116,5 +106,69 @@ public async Task<Result<DownloadAttachmentDto>> DownloadAttachmentAsync(long at
...
+                ".msh1xml",".msh2xml",".scf",".lnk",".inf",".reg",".doc",".xls",
+                ".ppt",".docm",".dotm",".xlsm",".xltm",".xlam",".pptm",".potm",
+                ".ppam",".ppsm",".sldm",".dll"
+            };
+
+            foreach (var file in fileCollection)
+            {
+                foreach (var extention in dangerousExtentions)
+                {
+                    if(file.FileName.ToLower().EndsWith(extention))
```
Comment (SerjKocherga) [inline CharlieBackend.Business/Services/AttachmentService.cs:146]: Add space between "if" and open brace.

File: CharlieBackend.Business/Services/Interfaces/IBlobService.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+﻿using System.IO;
+using System.Threading.Tasks;
```
Comment (SerjKocherga) [inline CharlieBackend.Business/Services/Interfaces/IBlobService.cs:2]: Fix using directives placement to the right with code conventions.

**Review by SerjKocherga — CHANGES_REQUESTED** [review]

**Review by BohdanBBG — APPROVED** [review]

**Review by SerjKocherga — APPROVED** [review]

**Review by x1mx1x — APPROVED** [review]

**Review by filinskyi — APPROVED** [review]

File: CharlieBackend.Core/DTO/Attachment/AttachmentDto.cs [hunk 4]
```diff
@@ -1,14 +1,24 @@
...
 
 namespace CharlieBackend.Core.DTO.Attachment
 {
     public class AttachmentDto
     {
+        [Required]
         public long Id { get; set; }
 
+        [Required]
+        public DateTime CreatedOn { get; set; }
```
Comment (Hlib-mats) [inline CharlieBackend.Core/DTO/Attachment/AttachmentDto.cs:13]: maybe you should add an attribute [DataType(DataType.Date)]?

**Review by Hlib-mats — APPROVED** [review]

File: CharlieBackend.Core/DTO/Attachment/AttachmentDto.cs (hunk 4 above)
Comment (SnipeFrost) [reply CharlieBackend.Core/DTO/Attachment/AttachmentDto.cs:13]: right

Comment (SnipeFrost) [reply CharlieBackend.Core/DTO/Attachment/AttachmentDto.cs:13]: but need [DataType(DataType.DateTime)] since we also need time

Comment (Hlib-mats) [reply CharlieBackend.Core/DTO/Attachment/AttachmentDto.cs:13]: you're right

**Review by Hlib-mats — APPROVED** [review]

File: CharlieBackend.Business/Services/AttachmentService.cs (hunk 1 above)
Comment (sober607) [inline CharlieBackend.Business/Services/AttachmentService.cs:131]: I think its settings of project, and better be somewhere at .json settings

**Review by sober607 — APPROVED** [review]

File: CharlieBackend.Business/Services/AttachmentService.cs (hunk 1 above)
Comment (SnipeFrost) [reply CharlieBackend.Business/Services/AttachmentService.cs:131]: Its not setting of project

**Pull request merged**